pkg/scheduler/reconciliation/inmemory.go

```diff
@@ -213,6 +213,10 @@ func (r *InMemoryReconciliationRepository) UpdateOperationState(schedulingID, co
 		return &repository.EntityNotFoundError{}
 	}
 
+	if err := operationAlreadyInState(op, state); err != nil && !allowInState {
```

Comment:
I still trying to understand why we need this allowInState, but for code readability improvement. I assume the following way is a bit easy to understand, what do you think? If not allowed, then you don't have to check the op.state and state, right?
and also better use directly equal check in condition, use `!allowInState` I would suggest to rename it to `notAllowInState`
```
		if allowInState {
			err := operationAlreadyInState(op, state)
			if err != nil {
				return err
			}
		}
```

Reply:
Already addressed in personal meeting